features/FEATURE_BLE/ble/pal/PalSecurityManager.h

```diff
@@ -0,0 +1,1017 @@
+     * passkey every time it calls SecurityManagerEvent::on_passkey_display.
+     *
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t set_display_passkey(
+        passkey_num_t passkey
+    ) = 0;
+
+    /**
+     * Reply to a passkey request received from the SecurityManagerEventHandler.
+     *
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t passkey_request_reply(
+        connection_handle_t connection,
+        passkey_num_t passkey
+    ) = 0;
+
+    /**
+     * Reply to an oob data request received from the SecurityManagerEventHandler.
+     *
+     * @param[in] connection connection handle
+     * @param[in] oob_data pointer to out of band data
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t legacy_pairing_oob_data_request_reply(
+        connection_handle_t connection,
+        const oob_tk_t &oob_data
+    ) = 0;
+
+    /**
+     * Notify the stack that the user has confirmed the values during numerical
+     * comparison stage of pairing.
+     *
+     * @param[in] connection connection handle
+     * @param[in] confirmation true if the user indicated the numbers match
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t confirmation_entered(
+        connection_handle_t connection,
+        bool confirmation
+    ) = 0;
+
+    /**
+     * Notify the stack that the user pressed a key. This will be sent to the peer and create
+     * an appropriate event there if the keypress protocol is enabled.
+     *
+     * @param[in] connection connection handle
+     * @param[in] keypress type of keypress event
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t send_keypress_notification(
+        connection_handle_t connection,
+        Keypress_t keypress
+    ) = 0;
+
+    /**
+     * Notify the stack that the OOB data has been verified and supply the peer's random number.
+     * If the verification failed this will not be called and cancel_pairing will be called instead.
+     *
+     * @param[in] connection connection handle
+     * @param[in] oob_rand_t random number sent from the local device to be used in further
+     *                       calculations by the stack, set to 0 if peer reported no OOB present
+     * @param[in] oob_rand_t random number from the peer to be used in further
+     *                       calculations by the stack, set to 0 if no OOB data received
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t oob_data_verified(
+        connection_handle_t connection,
+        const oob_rand_t &local_random,
+        const oob_rand_t &peer_random
```

Comment:
0 is a valid random number

Reply:
true but not relevant to the process - that is what the spec dictates page 2324 of v5